src/transaction/locator_sr.c

```diff
@@ -13937,6 +13941,166 @@ has_errors_filtered_for_insert (std::vector<int> error_filter_array)
 }
 // *INDENT-ON*
 
+
+static inline void
+locator_initialize_classname_entry (LOCATOR_CLASSNAME_ENTRY * entry, const char *classname, const OID * classoid)
+{
+  assert (classname != nullptr);
+
+  /* Memory for classname is allocated in atomic_replicator::update_classname_cache_for_ddl () */
+  entry->e_name = (char *) classname;
```

Comment:
Shouldn't it be copied?
```suggestion
      entry->e_name = strdup ((char *) classname);
      if (entry->e_name == NULL)
```

Reply:
I used `strdup` at initial commit. However, since the classname was also allocated in the caller function, I attempted to use an already allocated heap memory to reduce the number of memory allocations and free. However, I am a bit concerned about the readability

Reply:
I'd use `strdup` not only for readability, but also the fact that a caller should know the passed string may be released somewhere. It's "moved" actually, and this should be warned by users of this function. I doubt it's worth it, so I'd just use `strdup`.

Reply:
OK, I agree. I will change it